ui/src/components/mobileservices/BindingPanel.js

```diff
@@ -46,20 +81,58 @@ export class BindingPanel extends Component {
     this.open();
   }
 
-  componentWillMount() {
-    const serviceName = this.props.service.getName();
-    const schema = this.props.service.getBindingSchema();
-    const form = this.props.service.getFormDefinition();
-    const { service } = this.props;
+  hasUPSAndroidAnnotation() {
+    return this.hasUPSPlatformAnnotation('android');
+  }
 
-    this.setState({
-      serviceName,
-      schema,
-      form,
-      loading: false,
-      service,
-      activeStepIndex: 0
-    });
+  hasUPSIOSAnnotation() {
+    return this.hasUPSPlatformAnnotation('ios');
+  }
+
+  hasUPSPlatformAnnotation(platform) {
+    // configExt field example value:
+    // it is an array of annotations that start with org.aerogear.binding-ext
+    // and our annotation's value is also an array
+    /*
+      [
+        [
+          {
+            "type": "android",
+            "typeLabel": "Android",
+            "url": "https://ups-mdc.127.0.0.1.nip.io/#/app/8936dead-7552-4b55-905c-926752c759af/variants/d6f4836a-11df-42d1-a442-e9cc823715a4",
+            "id": "d6f4836a-11df-42d1-a442-e9cc823715a4"
+          }
+        ]
+      ]
+      */
+    // there won't be any variant annotations if there is no binding yet
+    if (!this.props.service.isBound()) {
+      return false;
+    }
+
+    const configExt = this.props.service.getConfigurationExt();
+
+    if (!configExt || !configExt.length) {
+      return false;
+    }
+
+    for (const configItemStr of configExt) {
+      let configExtItem;
+      try {
+        configExtItem = JSON.parse(configItemStr);
```

Comment:
probably can move this to be a new method in the `MobileServicce` model, call it `getConfigurationExtAsJSON` or something similar.

Reply:
ok, sure let me do it